DEVELOPMENT.md

```diff
@@ -0,0 +1,536 @@
+
+```bash
+cd frontend
+
+# Install dependencies (first time only)
+npm install
+
+# Start with local config
+npm run dev -- --config vite.config.test.js --port 5173
+
+# Start with production config
+npm run dev -- --config vite.config.js --port 5173
+```
+
+**Default URLs:**
+- Frontend: http://localhost:5173
+- API (local): http://localhost:8001
+- API (production): https://nyrk.io
+
+### Full Stack Development
+
+For a complete local development environment:
+
+```bash
+# Terminal 1: Start backend
+python3 etc/nyrkio_backend.py start
+
+# Terminal 2: Start frontend in local mode
+python3 etc/nyrkio_frontend.py start
+
+# Or run both and check status
+python3 etc/nyrkio_backend.py status
+python3 etc/nyrkio_frontend.py status
+```
+
+## Running the Backend
+
+### Using nyrkio_backend.py (Recommended)
+
+The `etc/nyrkio_backend.py` script manages the backend server process:
+
+```bash
+# Start the backend server
+python3 etc/nyrkio_backend.py start
+
+# Check if backend is running
+python3 etc/nyrkio_backend.py status
+
+# Stop the backend server
+python3 etc/nyrkio_backend.py stop
+
+# Restart the backend server
+python3 etc/nyrkio_backend.py restart
+```
+
+**Features:**
+- Runs server in background
+- Auto-detects Poetry or system Python environment
+- Manages PID file for process tracking
+- Graceful shutdown with fallback to force kill
+
+**Default Configuration:**
+- Port: 8001 (configurable via `.env.backend`)
+- Auto-reload: Enabled for development
+
+### Using Poetry Directly
+
+For foreground execution with full output:
+
+```bash
+cd backend
+poetry run uvicorn backend.api.api:app --host 0.0.0.0 --port 8001 --reload
+```
+
+### Using Docker Compose (Full Stack)
+
+To run the complete stack including nginx, webhooks, and MongoDB:
+
+```bash
+# Start Docker stack
+python3 etc/nyrkio_docker.py start
+
+# Check Docker stack status
+python3 etc/nyrkio_docker.py status
+
+# Stop Docker stack
+python3 etc/nyrkio_docker.py stop
+
+# Restart Docker stack
+python3 etc/nyrkio_docker.py restart
+
+# Or using docker compose directly
+export IMAGE_TAG=$(git rev-parse HEAD)
+docker compose -f docker-compose.dev.yml up --build
+```
+
+**Services:**
+- Backend API: http://localhost:8000
+- Webhooks: http://localhost:8080
+- Nginx proxy: http://localhost:80
+- MongoDB: localhost:27017
+
+## Testing Management Scripts
+
+To verify that all management scripts are working correctly:
+
+```bash
+# Run the management scripts test suite
+python3 etc/test_management_scripts.py
+```
+
+This test suite verifies:
+- All scripts exist and are executable
+- Scripts respond to `--help` flag
+- Scripts accept correct command-line arguments
+- Status commands work when services are stopped
+- Vite configuration files exist and have correct targets
+- Documentation references correct script paths
+- No references to old script names remain
+
+## Running Tests
+
+### Backend Tests
+
+The `backend/runtests.sh` script provides various testing options:
```

Comment:
Should this be converted to a similar python script as the ones above? Can be a separate followup task of course.